### PR TITLE
Instance eliminate module naming optimization

### PR DESCRIPTION
**Pull request by pengzhengfa**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by pengzhengfa** [comment]
@KomachiSion Hello, please check

**Comment by KomachiSion** [comment]
PMD check failed, please run `mvn clean install -DskipITs -Dmaven.test.skip=true` in local and fix it.

**Comment by pengzhengfa** [comment]
ok,thank

**Comment by KomachiSion** [comment]
@pengzhengfa, Sorry there are a little conflict in PR.

File: naming/src/main/java/com/alibaba/nacos/naming/core/Cluster.java [hunk 1]
```diff
@@ -300,21 +300,21 @@ public void updateIPs(List<Instance> ips, boolean ephemeral) {
         return new ArrayList<>(instanceMap.values());
     }
 
-    public List<Instance> subtract(Collection<Instance> a, Collection<Instance> b) {
-        Map<String, Instance> mapa = new HashMap<>(b.size());
-        for (Instance o : b) {
-            mapa.put(o.getIp() + ":" + o.getPort(), o);
+    public List<Instance> subtract(Collection<Instance> oldIp, Collection<Instance> ips) {
```
Comment (KomachiSion) [inline naming/src/main/java/com/alibaba/nacos/naming/core/Cluster.java:303]: In this line, I had changed it to a private function. So there is a conflict.
Please change it.

Comment (pengzhengfa) [reply naming/src/main/java/com/alibaba/nacos/naming/core/Cluster.java]: @KomachiSion Hello, it has been solved. Please check it

**Pull request merged**